docs/src/content/docs/introduction/http-signatures.mdx

```diff
@@ -4,57 +4,122 @@ title: HTTP message signatures
 
 For more information about the EdDSA and its variants, refer to <LinkOut href='https://datatracker.ietf.org/doc/html/rfc8032'>RFC8032</LinkOut>.
 
+## Signature creation
+
+The example below illustrates creating the signature, starting with the original HTTP message.
+
+<CodeBlock title="Example: message before signature">
+
+```http
+POST HTTP/1.1
+Host: example.com
+Content-Type: application/json
+Content-Digest: sha-512=:X48E9qOokqqrvdts8nOJRJN3OWDUoyWxBf7kbu9DBPE=:
+Content-Length: 18
+Authorization: GNAP 123454321
+
+{
+    "hello";"world"
+}
+```
+
+</CodeBlock>
+
+![Create the http signature](/img/http-sig/4-sig_creation.png)
+
+### Signature Base
+
+The signature creation process begins with identifying the fields of the original message to use when creating the signature.
+These are called the covered components of the message. The covered components make up the signature base, together with the signing algorithm, and an identifier for the signer's public key.
+
+The final sub-field of the signature base is an HTTP structured field called signature params, an ordered list of components that make up the signature base.
+
 {/* prettier-ignore */}
-{/* ### Message signature creation
+{/*
 
-:::caution
-This section is WIP
-:::
+# fyi: contents of codeblock example below were generated using: https://httpsig.org
 
-### Bringing it all together
+\*/}
 
-:::caution
-This section is WIP
+<CodeBlock title="Example: signature base">
+
+```http
+"content-type": application/json
+"content-digest": sha-512=:X48E9qOokqqrvdts8nOJRJN3OWDUoyWxBf7kbu9DBPE=:
+"content-length": 18
+"authorization": GNAP 123454321
+"@method": POST
+"@target-uri": https://example.com/
+"@signature-params": ("content-type" "content-digest" "content-length" "authorization" "@method" "@target-uri");alg="ed25519";keyid="eddsa_key_1";created=1704722601
+
+```
+
+</CodeBlock>
+
+### Generate Signature
+
+To generate the http signature:
+
+- The signature base gets hashed (using SHA-512), producing a digest.
+- The digest gets signed with the signer's private key, producing the signature as a byte string.
+- The byte string gets Base64 encoded, and this results in the final signature value.
+
+### Signed HTTP Message
+
+The original message gets signed by adding uniqeuly labelled signature headers to the original message: `Signature-Input` and `Signature`.
+
+<CodeBlock title="Example: signed message">
+
+```http
+POST HTTP/1.1
+Host: https://example.com
+Content-Type: application/json
+Content-Length: 18
+Authorization: "GNAP 123454321"
+Signature-Input: sig1=("content-type" "content-digest" "content-length" "authorization" "@method" "@target-uri");alg="ed25519";keyid="eddsa_key_1";created=1704722601
+Signature: sig1=:EiCdZMbyXj6pN59g+mh3mY/Q6DlSBrCL7CJM4OZ550+d2MZhfdDKrOJU/ugeRdwd1KYyd1wA/VA7J2fi9YehCA==:
+
+{
+    "hello";"world"
+}
+```
+
+</CodeBlock>
+
+:::note
```

Comment:
When I look at this note in the preview, there's a line break between the two sentences. Can you either add a space between the lines or include them together in a single paragraph? Minor thing, I know.